contracts/core/src/libraries/XBlockMerkleProof.sol

```diff
@@ -9,6 +9,10 @@ import { XTypes } from "./XTypes.sol";
  * @dev Library for verifying XBlock merkle proofs
  */
 library XBlockMerkleProof {
+    uint8 internal constant DST_XBLOCK_HEADER = 1;
+
+    uint8 internal constant DST_XMSG = 2;
+
```

Comment:
Let's add some natspec comments here.

Also thinking - these aren't exaclty "domain separators" (is DST == domain separator? or domain separator type?)  They are kinda like the type hashes used in EIP-712 domain separators. But they also aren't type hashes. 

Is there another name we can give these? like `_TYPE_ID`? or maybe `_HASH_PREFIX`?

Reply:
Comments added.

I think these are domain separators (DST = domain separation tag), unless I'm very mistaken.

Reply:
I guess my understanding of the semantics of "domain separators" was that they were more about verifying chain & contract. but types are a domain in themselves. so this works